static/app/views/monitors/components/overviewTimeline/utils.tsx

```diff
@@ -2,39 +2,54 @@ import moment from 'moment';
 
 import {getFormat} from 'sentry/utils/dates';
 
-import {TimeWindow, TimeWindowData} from './types';
-
-// Stores options and data which correspond to each selectable time window
-export const timeWindowConfig: TimeWindowData = {
-  '1h': {
-    cursorLabelFormat: getFormat({timeOnly: true, seconds: true}),
-    elapsedMinutes: 60,
-    timeMarkerInterval: 10,
-    dateTimeProps: {timeOnly: true},
-  },
-  '24h': {
-    cursorLabelFormat: getFormat({timeOnly: true}),
-    elapsedMinutes: 60 * 24,
-    timeMarkerInterval: 60 * 4,
-    dateTimeProps: {timeOnly: true},
-  },
-  '7d': {
-    cursorLabelFormat: getFormat(),
-    elapsedMinutes: 60 * 24 * 7,
-    timeMarkerInterval: 60 * 24,
-    dateTimeProps: {},
-  },
-  '30d': {
-    cursorLabelFormat: getFormat({dateOnly: true}),
-    elapsedMinutes: 60 * 24 * 30,
-    timeMarkerInterval: 60 * 24 * 5,
-    dateTimeProps: {dateOnly: true},
-  },
+import {TimeWindow, TimeWindowOptions} from './types';
+
+// Stores the elapsed minutes for each selectable resolution
+export const resolutionElapsedMinutes: Record<TimeWindow, number> = {
+  '1h': 60,
+  '24h': 60 * 24,
+  '7d': 60 * 24 * 7,
+  '30d': 60 * 24 * 30,
 };
 
 export function getStartFromTimeWindow(end: Date, timeWindow: TimeWindow): Date {
-  const {elapsedMinutes} = timeWindowConfig[timeWindow];
-  const start = moment(end).subtract(elapsedMinutes, 'minute');
+  const start = moment(end).subtract(resolutionElapsedMinutes[timeWindow], 'minute');
 
   return start.toDate();
 }
+
+// The pixels to allocate to each time label based on (MMM DD HH:SS AM/PM)
+const TIMELABEL_WIDTH = 100;
+
+export function getConfigFromTimeRange(
+  start: Date,
+  end: Date,
+  timelineWidth: number
+): TimeWindowOptions {
+  // Acceptable intervals between time labels, in minutes
+  const minuteRanges = [1, 10, 30, 60, 4 * 60, 8 * 60, 12 * 60];
+  const startEndMinutes = (end.getTime() - start.getTime()) / (1000 * 60);
+  const timeLabelMinutes = startEndMinutes * (TIMELABEL_WIDTH / timelineWidth);
+
+  for (const minutes of minuteRanges) {
+    if (minutes > timeLabelMinutes) {
+      return {
+        cursorLabelFormat: getFormat({timeOnly: true}),
+        elapsedMinutes: startEndMinutes,
+        timeMarkerInterval: minutes,
+        dateTimeProps: {timeOnly: true},
+        tooltipDateTimeProps: {timeOnly: true},
```

Comment:
Let's just have one prop called `dateLabelFormat` and we can use that for the cursor and the tooltip